frontend/public/components/marketplace/modeless-manager.jsx

```diff
@@ -0,0 +1,97 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+
+import { Button, Modal } from 'patternfly-react';
+import { CatalogTile, CatalogTileBadge, CatalogItemHeader, PropertiesSidePanel, PropertyItem } from 'patternfly-react-extensions';
+
+import { mockPropertyItems } from './mockItems';
+
+class MarketplaceModelessOverlay extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { 
+      showOverlay: false 
+    };
+  }
+  close = () => {
+    this.setState({ 
+      showOverlay: false 
+    });
+  };
+  toggleOpen = () => {
+    this.setState({ 
+      showOverlay: !this.state.showOverlay 
+    });
```

Comment:
This should be `this.setState( prevState => { showOverlay: !prevState.showOverlay });`
React docs says it's more reliable that way.